gateway/docs/testing.md

```diff
@@ -0,0 +1,177 @@
+# Unit tests for gateway
+
+Gateway is written in [TypeScript](https://www.typescriptlang.org/) for a NodeJS environment. 
+The majority of block chain SDKs we use have type annotations for TypeScript so they integrate 
+smoothly into this project.
+
+Our main tool for unit testing is [jest](https://jestjs.io). Our configurations
+for jest are located [here](../jest.config.js).
+
+We use [prettier](https://prettier.io/) for formatting. Configuration file
+[here](../.prettierrc).
+
+We use [ESLint](https://eslint.org/) for linting. Configuration file 
+[here](../.eslintrc.js).
+
+## Running the unit tests
+
+We primarily use `yarn` for development, but `npm` should work as well.
+
+First install dependencies:
+
+```
+yarn
+```
+
+Then build the gateway package:
+
+```
+yarn build
+```
+
+Then you can run the unit tests:
+
+```
+yarn test:unit
+```
+
+You do not actually need to build before running, but if you have any compiler
+errors, the unit tests will crash, so in practice if you make changes, it is
+best to run build first.
+
+## Run individual unit test file
+
+`yarn test:unit` runs all of the tests. Alternatively, you can run the unit tests
+from a single file with `yarn jest path/to/test`. For example:
+
+```
+yarn jest test/chains/ethereum/ethereum.controller.test.ts
+```
+
+## Test coverage
+
+Our github repository has requirements for test coverage. These requirements may 
+change, so specifically look at our [workflow file](../../.github/workflows/workflow.yml) 
+and find the section for gateway unit tests. If you are interested in merging a PR,
+you will need to make sure unit test coverage meets the requirements.
+
+To see the current coverage, run the following:
+
+```
+yarn test:cov
+```
+
+This will produce a table with absolute numbers, percentages and line numbers that
```

Comment:
I think it's worth mentioning types of coverage requirement, e.g. %branch, %line, on what they are (or where we can read more about it). I didn't know what a %brach for until I googled it.